packages/frontend/src/lib/pnl.ts

```diff
@@ -38,53 +42,117 @@ export function calcShortGain({ shortUnrealizedPNL, usdAmount, wethAmount, ethPr
-  sellQuote,
-  wethAmount,
-  ethPrice,
-}: {
-  sellQuote: BigNumber
-  wethAmount: BigNumber
-  ethPrice: BigNumber
-}) {
-  return {
-    usdValue: sellQuote.minus(wethAmount.abs()).multipliedBy(ethPrice),
-    ethValue: sellQuote.minus(wethAmount.abs()),
-  }
-}
 
-export function calcETHCollateralPnl(
+export async function calcETHCollateralPnl(
   data: VaultHistory_vaultHistories[] | undefined,
-  ethPriceMap: { [key: string]: number },
   ethPrice: BigNumber,
+  currentEthBalance: BigNumber,
 ) {
-  const deposits = data?.length
-    ? data?.reduce((acc: BigNumber, curr: VaultHistory_vaultHistories) => {
-        const time = new Date(Number(curr.timestamp) * 1000).setUTCHours(0, 0, 0) / 1000
+  const { deposits, withdrawals } = data?.length
+    ? await data?.reduce(async (prevPromise, curr: VaultHistory_vaultHistories) => {
+        const acc = await prevPromise
+
+        const dateTimeString = getDateStringForPnl(curr.timestamp)
+        const historicEthPrice = await getHistoricEthPrice(dateTimeString)
+
         if (curr.action === Action.DEPOSIT_COLLAT) {
-          acc = acc.plus(
-            new BigNumber(toTokenAmount(curr.ethCollateralAmount, 18)).times(
-              new BigNumber(ethPrice).minus(ethPriceMap[time]),
-            ),
-          )
-        }
-        return acc
-      }, new BigNumber(0))
-    : new BigNumber(0)
-
-  const withdrawals = data?.length
-    ? data?.reduce((acc: BigNumber, curr: VaultHistory_vaultHistories) => {
-        const time = new Date(Number(curr.timestamp) * 1000).setUTCHours(0, 0, 0) / 1000
-        if (curr.action === Action.WITHDRAW_COLLAT) {
-          acc = acc.plus(
-            new BigNumber(toTokenAmount(curr.ethCollateralAmount, 18)).times(
-              new BigNumber(ethPrice).minus(ethPriceMap[time]),
-            ),
+          acc.deposits = historicEthPrice
+            ? acc.deposits.plus(new BigNumber(toTokenAmount(curr.ethCollateralAmount, 18)).times(historicEthPrice))
+            : BIG_ZERO
+        } else if (curr.action === Action.WITHDRAW_COLLAT) {
+          acc.withdrawals = acc.withdrawals.plus(
+            new BigNumber(toTokenAmount(curr.ethCollateralAmount, 18)).times(historicEthPrice),
           )
         }
         return acc
-      }, new BigNumber(0))
-    : new BigNumber(0)
+      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO }))
+    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO }
+
+  return currentEthBalance.times(ethPrice).isEqualTo(0) || deposits.minus(withdrawals).isEqualTo(0)
+    ? BIG_ZERO
+    : currentEthBalance.times(ethPrice).minus(deposits.minus(withdrawals))
+}
+
+export async function calcDollarShortUnrealizedpnl(
+  swaps: swaps_swaps[],
+  isWethToken0: boolean,
+  buyQuote: BigNumber,
+  ethPrice: BigNumber,
+) {
+  const { totalWethInUSD } = await swaps.reduce(async (prevPromise, curr: any) => {
+    const acc = await prevPromise
+    const squeethAmt = new BigNumber(isWethToken0 ? curr.amount1 : curr.amount0)
```

Comment:
what is curr? the current swap the user has? I guess the var name is a bit confusing, interpreting it as current but then historicEthPrice being based on curr.timestamp wouldn't really make sense?